mobilecoind/src/payments.rs

```diff
@@ -1762,6 +1765,38 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_select_utxos_for_optimization_works_with_eusd() {
```

Comment:
Not understanding how this test pertains to the code change?

The code change "normalized" the inputs being passed to `select_utxos_for_optimization()` to be from the same token id. This test seems to be testing `select_utxos_for_optimization()`, but not the result of the input normalization.

Reply:
I'm just trying to follow the patterns that exist and add test coverage where it doesn't exist.

If it were easy to construct a TransactionManager instance in a test, we could do that, but it's rather complicated, what with connection manager and all that, and I don't see any example to go off of.

If you prefer not to have this test I'm happy to remove it.

Reply:
I guess I kind of glanced over the test name it does say "works with eusd"
From that perspective I can see how this improves case coverage.
I've approved it will need some changes to actually compile.